File: 435 (Non-overlapping Intervals)/javascript/Solution.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
  if (intervals.length<=1) return 0;
  let res = 0;

  intervals.sort((a,b) => a[0]-b[0]);
  // console.log(intervals);
  let prev = intervals[0];
  for(let i =1;i<intervals.length;i++){
      let curr= intervals[i];
      if (prev[1]>curr[0]){
          res++;
          if (prev[1]>curr[1]){
              prev = curr;
          }
      }else{
          prev = curr;
      }
  }
  return res;
};

File: 435 (Non-overlapping Intervals)/python/Solution.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        if (len(intervals)) <=1: return 0

        res = 0
        intervals.sort(key = lambda x: x[0])

        prev = intervals[0]
        for curr in intervals[1:]:
            if prev[1]>curr[0]:
                res+=1

                if prev[1]>curr[1]:
                    prev = curr

            else:
                prev = curr
        return res